website/docs/opengl/c1/3-hello-texture.md

```diff
@@ -0,0 +1,447 @@
+
+In your fragment shader, add this `uniform` declaration:
+```
+uniform sampler2D uTexture;
+```
+
+And change this line:
+```
+void main()
+{
+//  -out_color = vec4(frag_texCoords.x, frag_texCoords.y, 0, 1.0);
+    out_color = texture(uTexture, frag_texCoords);
+}
+```
+
+In GLSL, you usually don't work with textures directly. The most common way to access a texture is using a `sampler`.
+In our case, we are using `sampler2D` for our 2D texture.
+To read the texture data, the `texture` method is used. It samples the color of the texture at the
+specified texture coordinates and uses the previously set parameters to interpolate that color.
+
+But pay attention! If you have good eyes, you noticed our texture coordinates are between the ranges of `0.0` to `1.0` This is because the `texture` method works
+with normalized values! To better understand this, think about a 250x500 pixels image. If you want to get the pixel at the position
+(250, 250), we need to send (250 / width, 250 / height), or (1, 0.5), as the texture coordinate. This way the size of the texture doesn't matter to the shader!
+
+<?# Info "You can use the equation ` 1/size * pixel_position ` to get the normalized coordinate for a particlar axis!" /?>
+
+Now, if you run the program, you will see just a black quad. It's because we need to bind the texture before the draw call!
+If you don't do it, the last applied texture will be used for this mesh, in our case, since we have no texture bound after the clean up, no texture is used.
+
+To solve it, go to your `OnRender` method and, below the `_gl.UseProgram()` call, first activate the texture unit that you will use:
+```c#
+_gl.ActiveTexture(TextureUnit.Texture0);
+```
+As explaned before, a texture unit is a space in memory that refers to the texture object, like a pointer. We need to first active this "pointer" to use it.
+
+After that we bind the texture again. Doing it after activating the texture unit will automatically attatch the texture at the texture unit 0, as a 2D texture.
+```c#
+_gl.BindTexture(TextureTarget.Texture2D, _texture);
+```
+With it, the texture in the texture unit 0 should be sampled for the 1° Sampler2D.
+
+
+And now when you run it (drumroll...), you can see the image being drawn inside the quad!
+
+![Quad with texture](../../../images/opengl/chapter1/lesson3/quad-with-texture.png)
+
+you can see the code final result [clicking here](../sources/1.3-final-result.html).
+
+## Transparency in OpenGL
+Well, you must have noticed the black rectangle around the texture. If you used another program to check the used texture, it's completely transparent!
+So why our render are drawing it like it's not?
+Basically because, for OpenGL eyes, it's not!
+
+Transparency is a really expensive task in computer graphics. This is not a reason for you not to use this feature, but is a reason for OpenGL don't enable it
+as default. OpenGL have different ways to handle transparency and it's expected tuser choose for that.
```

Comment:
```suggestion
Transparency is a really expensive task in computer graphics. This is not a reason for you to avoid this feature, but is a reason that OpenGL doesn't enable it
by default. OpenGL has various different ways to handle transparency and it's expected that the user configures them explicitly.
```